addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/httputils/HtmlContextAnalyser.java

```diff
@@ -324,9 +324,16 @@ public List<HtmlContext> getHtmlContexts(
                 Iterator<Attribute> iter = element.getAttributes().iterator();
                 while (iter.hasNext()) {
                     Attribute att = iter.next();
+
                     if (att.getValue() != null
-                            && att.getValue().toLowerCase().indexOf(target.toLowerCase()) >= 0) {
+                            && att.getValue().toLowerCase().indexOf(target.toLowerCase()) >= 0
+                            && context.getStart() >= att.getValueSegment().getBegin()
+                            && context.getEnd() <= att.getValueSegment().getEnd()) {
                         // Found the injected value
+                        if (!context.getSurroundingQuote().equals("" + att.getQuoteChar())
+                                && !("" + att.getQuoteChar()).equals(" ")) {
```

Comment:
`&& att.getQuoteChar() != ' '`, this should be covered in `HtmlContextAnalyserUnitTest`.